libr/io/p/io_self.c

```diff
@@ -546,48 +546,34 @@ bool bsd_proc_vmmaps(RIO *io, int pid) {
 		}
 		ut8 *p_start = p;
 		ut8 *p_end = p + size;
-		int n = 0;
-
-		while (p_start < p_end) {
-			struct kinfo_vmentry *entry = (struct kinfo_vmentry *)p_start;
-			size_t sz = entry->kve_structsize;
-			if (sz == 0) {
-				break;
-			}
-			p_start += sz;
-			n ++;
-		}
-
-		struct kinfo_vmentry *entries = calloc(n, sizeof(*entries));
-		if (!entries) {
-			eprintf ("entries allocation failed\n");
-			goto exit;
-		}
-
-		p_start = p;
 
 		while (p_start < p_end) {
 			struct kinfo_vmentry *entry = (struct kinfo_vmentry *)p_start;
 			size_t sz = entry->kve_structsize;
 			int perm = 0;
+			char pstr[4] = {0};
```

Comment:
this wont  work because initial  string  should be "---". anyway this is implemented in r_str_rwx, better use this function